schema/messaging.proto

```diff
@@ -0,0 +1,463 @@
+
+// A chat room.
+message Room {
+  // The URI of the chat room.
+  string name = 1 [(google.api.resource).path = "rooms/{room_id}"];
+
+  // The human readable name of the chat room.
+  string display_name = 2 [(google.api.field_behavior) = REQUIRED];
+
+  // The description of the chat room.
+  string description = 3;
+
+  // The timestamp at which the room was created.
+  google.protobuf.Timestamp create_time = 4 [
+    (google.api.field_behavior) = OUTPUT_ONLY
+  ];
+
+  // The latest timestamp at which the room was updated.
+  google.protobuf.Timestamp update_time = 5 [
+    (google.api.field_behavior) = OUTPUT_ONLY
+  ];
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\CreateRoom
+// method.
+message CreateRoomRequest {
+  // The room to be created. The "room.name", "room.create_time", and
+  // "room.update_time" fields will be ignored by the server in this request.
+  Room room = 1;
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\GetRoom
+// method.
+message GetRoomRequest {
+  // The URI of the requested room.
+  string name = 1 [
+    (google.api.resource_reference) = "google.showcase.v1alpha3.Room",
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\UpdateRoom
+// method.
+message UpdateRoomRequest {
+  // The room to be updated.
+  Room room = 1;
+
+  // The field mask to determine wich fields are to be updated. If empty, the
+  // server will assume all fields are to be updated.
+  google.protobuf.FieldMask update_mask = 2;
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\DeleteRoom
+// method.
+message DeleteRoomRequest {
+  // The URI of the requested room.
+  string name = 1 [
+    (google.api.resource_reference) = "google.showcase.v1alpha3.Room",
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\ListRooms
+// method.
+message ListRoomsRequest {
+  // The maximum number of rooms to be returned. Server may return fewer rooms
+  // than requested. If unspecified, server will pick an appropriate default.
+  int32 page_size = 1;
+
+  // Typically, this is the value of
+  // google.showcase.v1alpha3.ListRoomsResponse.next_page_token
+  // returned from the previous call to
+  // `google.showcase.v1alpha3.Messaging\ListRooms` method.
+  string page_token = 2;
+}
+
+// The response message for the google.showcase.v1alpha3.Messaging\ListRooms
+// method.
+message ListRoomsResponse {
+  // The list of rooms.
+  repeated Room rooms = 1;
+
+  // A token to retrieve next page of results.
+  // Pass this value in ListRoomsRequest.page_token field in the subsequent
+  // call to `google.showcase.v1alpha3.Message\ListRooms` method to retrieve the
+  // next page of results.
+  string next_page_token = 2;
+}
+
+// This protocol buffer message represents a message sent to a chat room or
+// posted on a user profile.
+message Message {
```

Comment:
I like `Blurb` especially since there's little naming overloads. Another possibility is `Statement` but that's also overloaded (albeit less overloaded than message). I'll try `Blurb`.

Reply:
Done!